[docs] Fix typos and grammar in getting started (#17880)
mui-org/material-ui: docs/src/pages/getting-started/faq/faq.md

To correct this issue, all components on the page need to be initialized such that there is only ever **one class name generator** between them.

> You could end up accidentally using two-class name generators in a variety of scenarios:

Reviewer: ```suggestion
You could end up accidentally using two class name generators in a variety of scenarios:
```

## Do I have to use JSS to style my app?

No, it's not required.
But this dependency comes built in, so carries no additional bundle size overhead.

> However, perhaps you're adding some Material-UI components to an app that already uses another styling solution,

Reviewer: ```suggestion
Perhaps, however,  you're adding some Material-UI components to an app that already uses another styling solution,
```

If you are seeing a warning message in the console like the one below, you probably have several instances of `@material-ui/styles` initialized on the page.

> It looks like there are several instances of `@material-ui/styles` initialized in this application.

> > This may cause theme propagation issues, broken class names, specificity issues, and makes your application bigger without a good reason.

Reviewer: ```suggestion
> This may cause theme propagation issues, broken class names, specificity issues, and make your application bigger without a good reason.
```

### Duplicated module in node_modules

> If you think that the issue is in the duplicated @material-ui/styles module somewhere in your dependencies, there are several ways to check this.

Reviewer: ```suggestion
If you think that the issue may be in the duplication of the @material-ui/styles module somewhere in your dependencies, there are several ways to check this.
```

## My App doesn't render correctly on the server

If it doesn't work, in 99% of cases it's a configuration issue.

> A missing property, a wrong call order, or a missing component. Server-side rendering is strict about configuration, and the best way to find out what's wrong is to compare your project to an already working setup, check out the [reference implementations](/guides/server-rendering/#reference-implementations), bit by bit.

Reviewer: ```suggestion
A missing property, a wrong call order, or a missing component – server-side rendering is strict about configuration, and the best way to find out what's wrong is to compare your project to an already working setup. Check out the [reference implementations](/guides/server-rendering/#reference-implementations), bit by bit.
```

With this simple heuristic `Portal` might re-render after it mounts because refs are up-to-date before any effects run.
However, just because a ref is up-to-date doesn't mean it points to a defined instance.
If the ref is attached to a ref forwarding component it is not clear when the DOM node will be available.

> In the above example, the `Portal` would run an effect once but might not re-render because `ref.current` is still `null`. This is

Reviewer: ```suggestion
In the example above, the `Portal` would run an effect once, but might not re-render because `ref.current` is still `null`. This is
```
